src/tsig/mod.rs

```diff
@@ -1311,23 +1333,29 @@ impl<'a, Octs: Octets + ?Sized> MessageTsig<'a, Octs> {
     /// Checks that there is exactly one TSIG record in the additional
     /// section, that it is the last record in this section. If that is true,
     /// returns the parsed TSIG records.
-    fn from_message(msg: &'a Message<Octs>) -> Option<Self> {
-        let mut section = msg.additional().ok()?;
+    fn from_message(msg: &'a Message<Octs>) -> Result<Self, TsigError> {
+        let mut section = msg.additional().map_err(|_| TsigError::Missing)?;
```

Comment:
Maybe this should actually return something that gets turned into a form error? Processing is going to fail later, anyway, so might just return the error early?